policyeval/execute.go

```diff
@@ -50,6 +50,13 @@ func (pe *PolicyEvaluator) ApplyPolicy(ctx context.Context, userID id.UserID, po
 			for _, room := range rooms {
 				pe.ApplyBan(ctx, userID, room, recs.BanOrUnban)
 			}
+			// Update relevant cached entries in the policy server
+			for evtID, cache := range pe.policyServer.EventCache {
+				if cache.PDU.Sender == userID && cache.PDU.RoomID == recs.BanOrUnban.RoomID {
```

Comment:
This seems to be missing locking, also should probably have a separate function doing this rather than manually looping in unrelated code